### PR TITLE
Refactor pipelines

### PR DESCRIPTION
**Pull request by abnegate**
- Add pint + phpstan as dev dependencies
- Add test + codeql github workflows
- Analysis fixes

### CODE REVIEW

File: src/System/System.php [hunk 1]
```diff
@@ -177,49 +157,37 @@ public static function isPPC(): bool
...
     public static function getCPUCores(): int
     {
         switch (self::getOS()) {
             case 'Linux':
-                $cpuinfo = file_get_contents('/proc/cpuinfo');
-                preg_match_all('/^processor/m', $cpuinfo, $matches);
+                $cpuInfo = file_get_contents('/proc/cpuinfo');
+                $matches[] = [];
+
+                if ($cpuInfo) {
```
Comment (fanatic75) [inline src/System/System.php:188]: Do we need to check if it's empty or null here?

File: src/System/System.php [hunk 2]
```diff
@@ -79,8 +79,6 @@ public static function getOS(): string
 
     /**
      * Returns the architecture of the system's processor.
-     *
-     * @return string
```
Comment (fanatic75) [inline src/System/System.php:83]: let us keep all the return comment docs from function declaration as functions are returning values

Comment (abnegate) [reply src/System/System.php]: Fixed

**Review by fanatic75 — COMMENTED** [review]
Let's run composer format command in lint job file

File: src/System/System.php [hunk 3]
```diff
@@ -496,8 +499,8 @@ public static function getIOUsage($duration = 1): array
         });
 
         $diskStat2 = array_filter($diskStat2, function ($disk) {
-            foreach (self::INVALIDDISKS as $filter) {
-                if (! isset($disk[2])) {
+            foreach (self::INVALID_DISKS as $filter) {
+                if (! isset($disk[2]) || !\is_string($disk[2])) {
```
Comment (fanatic75) [inline src/System/System.php:503]:
```suggestion
                if (!isset($disk[2]) || !\is_string($disk[2])) {
```

File: src/System/System.php [hunk 4]
```diff
@@ -513,8 +516,21 @@ public static function getIOUsage($duration = 1): array
 
         // Compute Delta
         foreach ($diskStat as $key => $disk) {
-            $stats[$key]['read'] = (((intval($diskStat2[$key][5]) - intval($disk[5])) * 512) / 1048576);
-            $stats[$key]['write'] = (((intval($diskStat2[$key][9]) - intval($disk[9])) * 512) / 1048576);
+            $read1 = $diskStat2[$key][5];
```
Comment (fanatic75) [inline src/System/System.php:519]: let's make this read2 as it is from second disk stats
```suggestion
            $read2 = $diskStat2[$key][5];
```

File: src/System/System.php [hunk 5]
```diff
@@ -513,8 +516,21 @@ public static function getIOUsage($duration = 1): array
...
+            $read2 = $disk[5];
```
Comment (fanatic75) [inline src/System/System.php:520]:
```suggestion
            $read1 = $disk[5];
```

File: src/System/System.php [hunk 6]
```diff
@@ -513,8 +516,21 @@ public static function getIOUsage($duration = 1): array
...
+
+            /**
+             * @phpstan-ignore-next-line
+             */
+            $stats[$key]['read'] = (((intval($read1) - intval($read2)) * 512) / 1048576);
+
+            /**
+             * @phpstan-ignore-next-line
+             */
+            $stats[$key]['write'] = (((intval($write1) - intval($write2)) * 512) / 1048576);
```
Comment (fanatic75) [inline src/System/System.php:533]:
```suggestion
            $write2 = $diskStat2[$key][9];
            $write1 = $disk[9];

            /**
             * @phpstan-ignore-next-line
             */
            $stats[$key]['read'] = (((intval($read2) - intval($read1)) * 512) / 1048576);

            /**
             * @phpstan-ignore-next-line
             */
            $stats[$key]['write'] = (((intval($write2) - intval($write1)) * 512) / 1048576);
```

File: src/System/System.php [hunk 7]
```diff
@@ -530,15 +546,19 @@ public static function getIOUsage($duration = 1): array
...
      *
      * @throws Exception
      */
-    public static function getNetworkUsage($duration = 1): array
+    public static function getNetworkUsage(int $duration = 1): array
     {
         // Create a list of interfaces
         $interfaces = scandir('/sys/class/net', SCANDIR_SORT_NONE);
 
+        if (! $interfaces) {
```
Comment (fanatic75) [inline src/System/System.php:558]:
```suggestion
        if (!$interfaces) {
```

File: composer.json [hunk 8]
```diff
@@ -16,19 +16,26 @@
...
+        "psr-4": {
+            "Utopia\\System\\": "src/System"
+        }
     },
     "scripts": {
-      "lint": "./vendor/bin/pint --test",
-      "format": "./vendor/bin/pint"
+        "lint": "./vendor/bin/pint --test",
+        "format": "./vendor/bin/pint",
+        "check": "./vendor/bin/phpstan analyse --level=max --memory-limit=1G src"
```
Comment (fanatic75) [inline composer.json:26]: are we only running phpstan on src directory or this covers everything?

**Review by fanatic75 — APPROVED** [review]
LGTM

**Pull request merged**